_posts/2017-07-25-access-tokens.md

```diff
@@ -19,7 +19,13 @@ An access token is generated by performing a POST request to the webstrate in qu
 request must contain a `token` field. The value of the `token` field will be used as the token
 duration in seconds. If the value is not a positive integer, it defaults to 300 seconds.
 
-A token request response will look similar to:
+This example code will create a token with a 300 second validity for the currently logged-in user when run in the developer tools console:
+
+```javascript
+(await fetch(document.location.pathname, { method: "POST", body: new URLSearchParams("token=300") }).then(response => response.json())).token
```

Comment:
Good idea with an example, but we should probably stick to one only convention within the example, e.g.

```javascript
const res = await fetch(document.location.pathname, {
  method: "POST", body: new URLSearchParams({ token: 300 })
});
const obj = await res.json();
console.log(obj);
```

Or:

```javascript
fetch(document.location.pathname, {
  method: "POST", body: new URLSearchParams({ token: 300 })
})
  .then(res => res.json())
  .then(obj => console.log(obj));
```

I think the rest of documentation uses the former convention, so it's probably better to stick to that.

Reply:
It’s true, the mixing of async and then is smelly, but I thought it was clearer and more concise than the alternatives.  The pure-await solution involves either assigning to temporary variables or doing something wonky like `await (await fetch(document.location.pathname, { method: "POST", body: new URLSearchParams("token=300") })).json();` (note the await (await …)).  A pure .then-based approach is bla aussi.  Bizarrely enough, the async+then felt most natural for this kind of concise approach.  _BUT_, this is the documentation, so explicit is probably better than concise, so I’ll go with something like your first version.